ethereum/statetransition/src/main/java/tech/pegasys/teku/statetransition/forkchoice/ForkChoice.java

```diff
@@ -157,20 +159,28 @@ private void processHead(Optional<UInt64> nodeSlot) {
           if (!result.isSuccessful()) {
             return SafeFuture.completedFuture(result);
           }
-          indexedAttestationProvider.getIndexedAttestations().stream()
-              .filter(
-                  attestation ->
-                      forkChoiceStrategy.contains(attestation.getData().getBeacon_block_root()))
-              .forEach(
-                  indexedAttestation ->
-                      forkChoiceStrategy.onAttestation(transaction, indexedAttestation));
           return transaction
               .commit()
               .thenRun(() -> updateForkChoiceForImportedBlock(block, result))
+              .thenRun(() -> applyVotesFromBlock(forkChoiceStrategy, indexedAttestationProvider))
```

Comment:
I don't think it makes much real difference, but it felt right to update the chain head and thus fire off events and potentially trigger generating attestations as soon as possible instead of waiting until all the attestations from the block have been processed.  It wasn't possible previously because the votes used the transaction that added the block.